### DISCUSSION TITLE
useRouteLoaderData not updating

### DISCUSSION THREAD

**Discussion by nathanaelphilip** [remix-run/remix#6031] [Q&A] [answered]
👋 

If I run `fetcher.load('/')` - this should basically run the loader that’s set on that root path, correct? And if so, should it not update the context for that loader so anything using `useRouteLoaderData`’s data would automatically update?

Or am I misunderstanding how these are connected?

**Comment by kiliman** [comment]
No. You access the data from fetchers with `fetcher.data`. `useLoaderData` and `useRouteLoaderData` are updated via navigation or revalidation.

**Reply by nathanaelphilip** [reply]
Thank you @kiliman; is there a way to revalidate a route dynamically? I’m trying to use the routers context to manage all my state, but it seems that may not be doable.

**Reply by kiliman** [reply, marked as answer]
https://reactrouter.com/en/main/hooks/use-revalidator

**Reply by nathanaelphilip** [reply]
It’s beautiful 😭😭😭😭😭
